clients/python/agentic-sandbox-client/README.md

```diff
@@ -2,91 +2,137 @@
 
 This Python client provides a simple, high-level interface for creating and interacting with sandboxes managed by the Agent Sandbox controller. It's designed to be used as a context manager, ensuring that sandbox resources are properly created and cleaned up.
 
-## Usage
+It supports a **scalable, cloud-native architecture** using Kubernetes Gateways and a specialized Router, while maintaining a convenient **Developer Mode** for local testing.
 
-### Prerequisites
+## Architecture
 
-- A running Kubernetes cluster (e.g., `kind`).
-- The Agent Sandbox controller must be deployed with the extensions feature enabled.
-- A `SandboxTemplate` resource must be created in the cluster.
-- The `kubectl` command-line tool must be installed and configured to connect to your cluster.
+The client operates in two modes:
 
-### Installation
+1.  **Production (Gateway Mode):** Traffic flows from the Client -> Cloud Load Balancer (Gateway) -> Router Service -> Sandbox Pod. This supports GKE Autopilot and high-scale deployments.
+2.  **Development (Tunnel Mode):** Traffic flows from Localhost -> `kubectl port-forward` -> Router Service -> Sandbox Pod. This requires no public IP and works on Kind/Minikube.
 
-1.  **Create a virtual environment:**
+## Prerequisites
+
+- A running Kubernetes cluster (GKE Autopilot recommended for production; Kind/Minikube supported for dev).
+- The **Agent Sandbox Controller** installed.
+- The **Sandbox Router** deployed in the cluster (see instructions below).
+- `kubectl` installed and configured locally.
+
+## Setup: Deploying the Router
```

Comment:
This is only needed for the Gateway Mode, right? This should be optional and under Gateway Mode installation.